djvm/src/main/kotlin/net/corda/djvm/code/EmitterModule.kt

```diff
@@ -202,6 +205,54 @@ class EmitterModule(
         hasEmittedCustomCode = true
     }
 
+    /**
+     * This determines which [sandbox.java.lang.Throwable] type we must up-cast
+     * the return value of [sandbox.java.lang.DJVM.catch] to.
+     */
+    fun commonThrowableClassOf(classNames: Iterable<String>): String {
+        val classIterator = classNames.iterator()
+        var throwable = classIterator.next()
+        while (classIterator.hasNext() && throwable != THROWABLE_NAME) {
```

Comment:
Stopping if `throwable == THROWABLE_NAME` seems reasonable, but what if part of the sequence after that does not descend from `Throwable`? It seems that in this case a `ClassCastException` might or not be thrown depending on the order of the elements.

Reply:
This function is iterating through the exception types found in a `catch` block. The byte-code verifier is going to reject anything here that does not descend from `Throwable`.